packages/reactivity/src/effectScope.ts

```diff
@@ -126,7 +126,11 @@ export function getCurrentScope() {
 
 export function onScopeDispose(fn: () => void) {
   if (activeEffectScope) {
-    activeEffectScope.cleanups.push(fn)
+    activeEffectScope.cleanups.push(() => {
+      pauseTracking()
+      fn()
+      resetTracking()
+    })
```

Comment:
I'm not very familiar with this part of the codebase, but I did have a couple of thoughts...

1. Is there a reason to call `pauseTracking()` and `resetTracking()` inside each cleanup function, rather than doing it once in `stop`, either side of the `for` loop that invokes these functions?
2. Do we need to handle error cases here? I notice that `run` uses `try`/`finally` for something similar. I'm thinking maybe `resetTracking()` should be in a `finally` block?

Reply:
Updated, thanks for your reivew.